### PR TITLE
Closes #6152: Add license type to what is reported to Mixpanel

### LINKED ISSUE #6152: Add license type to what is reported to Mixpanel
This will help see how representative is the current sample from opted-in users vs our current users and how reliable data collected from them to help with product decisions.

### PR DESCRIPTION
**Pull request by Miraeld**
## Description

This will help see how representative is the current sample from opted-in users vs our current users and how reliable data collected from them to help with product decisions.

A new function has been created that returns the type of license. As this functionality is used in different files, it has been modified to use the newly created function.

Fixes #6152 

## Type of change

- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

Yes

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [x] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

File: inc/functions/admin.php [hunk 1]
```diff
@@ -526,3 +526,33 @@ function rocket_create_options_hash( $value ) {
...
 	return md5( wp_json_encode( $value_diff ) );
 }
+
+/**
+ * This function returns the license type for a customer.
+ *
+ * @param object $customer_data
+ * @return string
+ */
+function get_license_type($customer_data) {
```
Comment (engahmeds3ed) [inline inc/functions/admin.php]: You are in the global namespace, you need to add a prefix here to be `rocket_get_license_key`

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]
Good job @Miraeld in ur first PR.

I didn't fully review it yet but I noticed this important point.

Also plz run the command
```
composer phpcs:fix
```

to automatically fix most of those reported coding standard issues.

File: inc/functions/admin.php (hunk 1 above)
Comment (Miraeld) [reply inc/functions/admin.php]: Oh alright, didn't know about that, I'm going to add it :)

Comment (Miraeld) [reply inc/functions/admin.php]: Alright, I've changed it :)

**Review by remyperona — CHANGES_REQUESTED** [review]
The code needs to be updated to follow our coding standards (errors are available in the CS GH action)

File: inc/functions/admin.php [hunk 2]
```diff
@@ -526,3 +526,31 @@ function rocket_create_options_hash( $value ) {
...
+	if ( false === $customer_data
+		|| ! isset( $customer_data->licence_account )
+	) {
+		return __( 'Unavailable', 'rocket' );
+	}
+
+	if ( 1 <= $customer_data->licence_account
+		&& $customer_data->licence_account < 3
+	) {
+		$type = 'Single';
```
Comment (engahmeds3ed) [inline inc/functions/admin.php]:
```suggestion
		return 'Single';
```

You can return directly and get rid of any else.

File: inc/functions/admin.php [hunk 3]
```diff
@@ -526,3 +526,31 @@ function rocket_create_options_hash( $value ) {
...
+ *
+ * @param object $customer_data customer data as an object.
+ * @return string
+ */
+function rocket_get_license_type( $customer_data ) {
+	$type = '';
+
+	if ( false === $customer_data
+		|| ! isset( $customer_data->licence_account )
+	) {
```
Comment (engahmeds3ed) [inline inc/functions/admin.php]: Let's make it more readable.

```suggestion
	if (
		false === $customer_data
		||
		! isset( $customer_data->licence_account )
	) {
```

File: inc/functions/admin.php [hunk 4]
```diff
@@ -526,3 +526,31 @@ function rocket_create_options_hash( $value ) {
...
+		return __( 'Unavailable', 'rocket' );
+	}
+
+	if ( 1 <= $customer_data->licence_account
+		&& $customer_data->licence_account < 3
+	) {
```
Comment (engahmeds3ed) [inline inc/functions/admin.php]:
```suggestion
	if (
		1 <= $customer_data->licence_account
		&&
		3 > $customer_data->licence_account
	) {
```

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]
Minor adjustments

File: inc/functions/admin.php (hunk 2 above)
Comment (Miraeld) [reply inc/functions/admin.php]: I've corrected it.

File: inc/functions/admin.php (hunk 3 above)
Comment (Miraeld) [reply inc/functions/admin.php]: I've corrected it.

File: inc/functions/admin.php (hunk 4 above)
Comment (Miraeld) [reply inc/functions/admin.php]: I've corrected it.

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by vmanthos** [comment]
@Miraeld Thank you for the PR! 
I see the `license_type` is added to the data sent to Mixpanel. :+1: 

@wp-media/productrocket The new info that will be collected is not listed here:
image

Also, the CTA (Activate Rocket Analytics) is displayed even when Rocket Analytics is enabled.

Please confirm (or not) that:
1. The license type info should be included in the data collection table.
2. The CTA should not be displayed if Rocket Analytics is already enabled.

**Comment by DahmaniAdame** [comment]
> The license type info should be included in the data collection table.

Yes, it should show. 

> The CTA should not be displayed if Rocket Analytics is already enabled.

No, there is no need to redisplay it if the consent is already given.

**Review by remyperona — APPROVED** [review]

**Comment by vmanthos** [comment]
@Miraeld Thank you for the fixes.

The CTA button is not displayed but with that, the following text is hidden as well:
image

Since that's a disclaimer notice it should stay in place.

Also, and this is a late notice on my part, there is inconsistency in how various info is displayed as far as the used fonts go:
image

This has to do with some of it being wrapped in `<code></code>` and that results in the use of a monospace font.

Unless there is a specific reason to have some part in `<code>`, maybe we can go with a uniform look there.

@DahmaniAdame What's your take on this?

**Comment by DahmaniAdame** [comment]
@vmanthos yes, uniformizing it is the way to go. 
`code` looks nicer, but not consistent and practical (doesn't look good when lines are wrapped). 
Uniformizing it without the `code` would make sense.

**Comment by Miraeld** [comment]
@vmanthos I've corrected the cta, now only the button is not displayed once rocket analytics is enabled. 

@vmanthos @DahmaniAdame I've taken off the `<code></code>` tags to standardize the formatting.

**Review by vmanthos — APPROVED** [review]
Thank you for the fixes, @Miraeld ! 

This is working as expected. :+1:

**Pull request merged**